Ref.Windows/ViewModels/WebSiteViewModel.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using Polsys.Ref.Models;
+
+namespace Polsys.Ref.ViewModels
+{
+    /// <summary>
+    /// The View Model for <see cref="WebSite"/>.
+    /// </summary>
+    internal class WebSiteViewModel : PublicationViewModelBase
+    {
+        // When adding properties, remember to add them to
+        // the reset and commit routines as well!
+        public DateTime AccessDate
+        {
+            get { return _accessDate; }
+            set
+            {
+                if (_accessDate != value)
+                {
+                    if (IsReadOnly)
+                        throw new InvalidOperationException("Not in edit mode.");
+
+                    _accessDate = value;
+                    NotifyPropertyChanged(nameof(AccessDate));
```

Comment:
To do: Extend the `SetProperty` method to take any type. There are now multiple properties like this.